### PR TITLE
pr-buddy: helm: support join_params in `teleport-kube-agent` chart

### PR DESCRIPTION
**Pull request by hugoShaka**
Internal PR for external contribution #14625.

I added a commit to the PR, doing a bunch of changes not necessarily related to its original scope but that were needed: more tests and changing parts of the docs.

Backporting the PR to v10 might be painful because of the kube-state changes, I'm not sure if this is necessary.

### CODE REVIEW

**Review by strideynet — APPROVED** [review]
Looks good, just one thought on security.

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx [hunk 1]
```diff
@@ -102,6 +105,67 @@ fail to join the Teleport cluster.
...
+
+### `joinParams.tokenName`
+| Type | Default value | Required? |
+| - | - | - |
+| `string` | `nil` | Yes |
+
+When `joinParams.method` is `iam` or `ec2`, `joinParams.tokenName` contains the name of the token that will be used to
+join the cluster. In this case, the value is not sensitive as authorization is checked through AWS IAM or EC2.
+
+When `joinParams.method` is `token` (by default), `joinParams.tokenName` contains the join token itself. In this case,
```
[the remark this answers is not in the document]
Comment (hugoShaka) [reply docs/pages/reference/helm-reference/teleport-kube-agent.mdx:148]: When I initially read the docs, I understood tokens were looked-up only when `method` was `token`, but I was not able to find this behaviour in our codebase. I don't have an AWS environment to quickly test if this works though.

**Review by tigrato — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx [hunk 2]
```diff
@@ -102,6 +105,77 @@ fail to join the Teleport cluster.
   </TabItem>
 </Tabs>
 
+## `joinParams`
+
+`joinParams` controls how the Teleport agent joins the Teleport cluster.
+Those sub-values must be configured for the agent to connect to its cluster.
```
Comment (alexfornuto) [inline docs/pages/reference/helm-reference/teleport-kube-agent.mdx:111]:
```suggestion
These sub-values must be configured for the agent to connect to a cluster.
```

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx [hunk 3]
```diff
@@ -102,6 +105,77 @@ fail to join the Teleport cluster.
...
+- For `iam`, see [Joining Nodes Via AWS IAM Role](../../management/guides/joining-nodes-aws-iam.mdx).
+- For `ec2`, see [Joining Nodes Via AWS IAM Role](../../management/guides/joining-nodes-aws-ec2.mdx).
+- For `token` (default value), the token must be provided through `joinParams.tokenName` or
+  [through an existing Kubernetes Secret](#secretName).
+
+<Admonition type="note" title="IAM joining requirements">
+Using the IAM joining method requires either the pods to have access to [instance
+metadata](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html),
+or [IAM roles for service
+account](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) to be
```
Comment (alexfornuto) [inline docs/pages/reference/helm-reference/teleport-kube-agent.mdx:130]:
```suggestion
accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) to be
```

**Review by alexfornuto — COMMENTED** [review]
Made some minor suggestions.

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (hugoShaka) [reply docs/pages/reference/helm-reference/teleport-kube-agent.mdx:158]: I finally got the time to build a full test setup and validate this works.

**Pull request merged**